aten/src/ATen/core/jit_type.h

```diff
@@ -1807,4 +1814,20 @@ static ScalarTypeTypePtr get();
 ScalarTypeType() : EnumerationType() {}
 };
 
+inline bool IValue::isDoubleList() const {
+  return isGenericList() && static_cast<detail::ListImpl*>(payload.as_intrusive_ptr)->elementType->isSubtypeOf(FloatType::get());
```

Comment:
Is it wrong to go through the IValue's type instead of the value for this query? e.g. `type()->expect<ListType>()->containedType()->cast<FloatType>() != nullptr`. Seems very marginally less icky.

Reply:
It is written this way to avoid the refcount bump that calling type() will cause to the containing type. Since this is on hot paths it seemed better to avoid it.